vtex-sites/nextjs.store
# feat: Working `SkuSelector` component

## What's the purpose of this pull request?

This PR implements the expected functionality for our SKU Selector component.

## How does it work?

Now the `ProductDetails` component will query for new fields in its `ProductDetailsFragment_product` query, namely:

- `activeVariations`
- `slugsMap`
- `availableVariations`

These new fields should bring all the information needed for us to place SKU Selectors on product pages.

They were added by this PR to `@faststore/api`: https://github.com/vtex/faststore/pull/1407.

## How to test it?

Since we didn't have any products with SKU properties configured that would enable the selector to function, in this branch, the deploy preview you'll find is pulling data from the `storecomponents` account instead of `storeframework`. So, to see the SKU Selector, go to a product such as [`Classic Shoes`](https://sfj-b770cd1--nextjs.preview.vtex.app/classic-shoes-37/p).

Here's a video demo: https://www.loom.com/share/8cf41e561ad54b88845e9effca2aeb8c.

## References

This PR depends on https://github.com/vtex/faststore/pull/1407.

## Checklist

<em>You may erase this after checking them all ;)</em>

**Changelog**

**PR Description**

**Documentation**
